csharp/test/Ice/defaultValue/AllTests.cs

```diff
@@ -284,23 +286,23 @@ public static void allTests(global::Test.TestHelper helper)
                     test(v.bs == null);
                     test(v.iseq == null);
                     test(v.st.a == 0);
-                    test(v.st2 != null);
+                    test(v.st2 == null);
```

Comment:
A struct-mapped-to-a-class field is considered non-nullable. The construction above is not correct since it sets st2 to null.

Prior to this PR, we would always create a class instance using the parameterless ctor.